### DISCUSSION TITLE
could not use python script

### DISCUSSION THREAD

**Discussion by includewins0ck2** [pawelsalawa/sqlitestudio#4911] [General]
Can't use the python plugin, I've tested it many times on a pure win11 system, no matter what version of python is installed, whether it's x86, or x64, with sqlitestudio 3.44 it all tells me It can't load the ScriptingPython.dll, but the file is clearly there.

**Comment by pawelsalawa** [comment]
Do you have Python installed in your system? Is it added to PATH?

**Comment by includewins0ck2** [comment]
> Do you have Python installed in your system? Is it added to PATH?

Yeah, I'm pretty sure. I installed a clean environment with vmware and tested it many times.

**Comment by tuffnatty** [comment]
Is it Python 3.9?

**Comment by includewins0ck2** [comment]
> Is it Python 3.9?
Thank you so much, I tried 3.8, 3.10, 3.11 but not 3.9, now it's solved, thanks!

**Comment by mx-lin** [comment]
Pawel, did you check python 3.12?
SQLiteStudio 3.4.4 portable, python 3.12, W11 (all latest versions) reports:

`
[13:53:18] Cannot load plugin ScriptingPython. Error details: Cannot load library D:\UTIL\SQLiteStudio\plugins\ScriptingPython.dll: The specified module could not be found.
`

The file is there:
```
Directory of D:\UTIL\SQLiteStudio\plugins
06.04.2023  22:54           118 272 ScriptingPython.dll
```
>

**Comment by grabani** [comment]
I am having the same issue.

This is where I try to enable the plugin...

![image](https://github.com/pawelsalawa/sqlitestudio/assets/25469154/5655671e-f0c0-4697-81e3-9dff5c12e944)

This is the error:

![image](https://github.com/pawelsalawa/sqlitestudio/assets/25469154/d8d5723e-aa58-418a-9518-02173dfbc358)

The DLL file exists in the location referenced in the error:
![image](https://github.com/pawelsalawa/sqlitestudio/assets/25469154/7fa3d2ea-ee8b-4a8f-86dd-f0ab63ee81e6)

I have python in my path and it is version 3.11.2

![image](https://github.com/pawelsalawa/sqlitestudio/assets/25469154/6d1a9ae1-1b07-45a3-9c46-6d419dfd1523)

**Comment by tuffnatty** [comment]
The binary builds at the moment are built to support only Python 3.9.

**Comment by jeffcwolf** [comment]
Just following up on this. I am using Mac OS Sonoma 14.6.1. I have the same problem and error when trying to load the ScriptingPython plugin. The error is:

[13:21:35] Cannot load plugin ScriptingPython. Error details: Cannot load library /Applications/SQLiteStudio/SQLiteStudio.app/Contents/PlugIns/libScriptingPython.dylib: (dlopen(/Applications/SQLiteStudio/SQLiteStudio.app/Contents/PlugIns/libScriptingPython.dylib, 0x000A): Library not loaded: libpython3.9.dylib
  Referenced from: <C09DA855-9CD3-35E4-AC71-B83766956236> /Applications/SQLiteStudio/SQLiteStudio.app/Contents/PlugIns/libScriptingPython.dylib
  Reason: tried: 'libpython3.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibpython3.9.dylib' (no such file), 'libpython3.9.dylib' (no such file), '/usr/local/lib/libpython3.9.dylib' (no such file), '/usr/lib/libpython3.9.dylib' (no such file, not in dyld cache), '//libpython3.9.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS//libpython3.9.dylib' (no such file), '//libpython3.9.dylib' (no such file), '/usr/local/lib/libpython3.9.dylib' (no such file), '/usr/lib/libpython3.9.dylib' (no such file, not in dyld cache))
  
  Now, I have confirmed - and reinstalled - Python 3.9 on my system and also placed in my PATH. For your reference, the default install location (via Homebrew) is /usr/local/bin/python3.9 (which is just a symbolic link that directs to /Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9)
  
  But I notice neither of these paths are listed in the error above exactly - the ''/usr/local/lib/libpython3.9.dylib'' is basically right, except the extension. But this is beyond my software knowledge...
  
  Any help would be appreciated! I want to use a Python script to automatically open files written as file paths in a data cell...
